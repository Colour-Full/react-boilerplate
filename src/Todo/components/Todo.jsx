// @flow
import * as React from 'react'
import type { Element } from 'react'

export type TodoProps = {
  onClick: () => void,
  completed: boolean,
  text: string
}

const Todo = ({onClick, completed, text}: TodoProps): Element<'li'> => (
  <li
    onClick={onClick}
    style={{
      textDecoration: completed ? 'line-through' : 'none'
    }}
  >
    {text}
  </li>
)

export default Todo
