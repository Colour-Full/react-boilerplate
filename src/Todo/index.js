// @flow
import React, { type Element } from 'react'

// Import Components
import Footer from './components/Footer'

// Import Containers
import AddTodo from './containers/AddTodo'
import VisibleTodoList from './containers/VisibleTodoList'


const Todo = (): Element<any> => (
  <div>
    <AddTodo />
    <VisibleTodoList />
    <Footer />
  </div>
)

export default Todo
