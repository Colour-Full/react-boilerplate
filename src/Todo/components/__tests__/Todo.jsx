// @flow

import React from 'react'
import Todo from '../Todo'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })

describe('A Link Component ', () => {
  it('It should render without throwing an error ', () => {
    const wrapper = shallow(<Todo />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should call onClick handler when <li /> is clicked', () => {
    const spy = jest.fn()
    const wrapper = shallow(<Todo onClick={spy} />)
    wrapper.find('li').simulate('click')
    expect(spy).toHaveBeenCalledTimes(1)
  })
})
