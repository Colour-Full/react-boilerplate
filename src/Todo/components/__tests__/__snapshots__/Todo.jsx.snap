// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Link Component  It should render without throwing an error  1`] = `
<li
  style={
    Object {
      "textDecoration": "none",
    }
  }
/>
`;
